### PR TITLE
Added cheat sheet listing functions used in the lessons

### PR DESCRIPTION
**Pull request by bojiutk**
Cheat sheet of functions used in the lessons

**Lesson 1 Introduction to R**

`length()` # how many elements are in a particular vector
`class() `# the class (the type of element) of an object
`str() `# an overview of the object and the elements it contains 
`c() ` # create vector; add elements to vector
` [  ] `    # extract and subset vector
`%in% ` # to test if a value is found in a vector
`is.na()`# test if there are missing values
`na.omit()` # Returns the object with incomplete cases removed
`complete.cases()`# elements which are complete cases

**Lesson 2 Starting with data**

`download.file() ` #download files from the internet to your computer 
`read.csv() ` # load CSV file into R memory 
`head() `# check the top (the first 6 lines) of an object including data frames
`factor() `# create factors
`levels() `# check levels of a factor
`nlevels() `# check number of levels of a factor
`as.numeric(levels()) ` # convert factors where the levels appear as numbers  to a numeric vector 

**Lesson 3 Introducing data.frame** 

`data.frame()`#create a data frame
`dim() ` #check dimension of data frame
`nrow() ` # returns the number of rows
`ncol() `  # returns the number of  columns 
`head() ` # shows the first 6 rows
`tail() ` #shows the last 6 rows
`names() ` #returns the column names (synonym of colnames() for data frame objects)
`rownames() ` # returns the row names
`str() ` # check structure of the object and information about the class, length and content of each column
`summary() `# summary statistics for each column
`seq() ` # generates a sequence of numbers

**Lesson 4 Aggregating and analyzing data with dplyr**

`install.packages()`# install a CRAN package in R
`library() ` # load installed package into the current session 
`select() ` # select columns of a data frame
`filter() `# allows you to select a subset of rows in a data frame 
`%>% ` # pipes to select and filter at the same time
`mutate() ` #create new columns based on the values in existing columns
`group_by() `# split the data into groups, apply some analysis to each group, and then combine the results.
`summarize() ` #collapses each group into a single-row summary of that group 
`tally()` # counts the total number of records for each category.
`write.csv() ` # save CSV file 

**Lesson 5 Data visualization with ggplot2**

`ggplot2(data=    , aes(x=  , y= ))+geom_point( )+facet_wrap ()+theme_bw()+theme() `
`aes()` #by selecting the variables to be plotted and the variables to define the presentation such as plotting size, shape color, etc.
`geom_` #graphical representation of the data in the plot (points, lines, bars). To add a geom to the plot use + operator
`facet_wrap()`# allows to split one plot into multiple plots based on a factor included in the dataset
`theme_bw()`#set the background to white
`theme()`# used to locally modify one or more theme elements in a specific ggplot object

**Lesson 6 R and SQL**

`dbConnect() `  # connect to the database
`dbGetQuery() ` # run the query and access data returned in a data frame object
`dbListTables() ` # list tables
`dbListFields() ` # get list of fields in the table

`dbDisconnect() `# Disconnect from the database
`dbWriteTable() ` # add the data frame to database

reference.md

### CODE REVIEW

File: reference.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+Cheat sheet of functions used in the lessons
+
+
+**Lesson 1**
+
+
+`length ( )` # how many elements are in a particular vector
```
Comment (fmichonneau) [inline reference.md]: I'm being picky, but could you please remove the spaces before the `(` and between the `()` in this function and all others below (e.g. `length()`) as it's the style we use throughout the lesson.

File: reference.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
+Cheat sheet of functions used in the lessons
+
+
+**Lesson 1**
```
Comment (fmichonneau) [inline reference.md]: could you please add the title of the lessons here and below (e.g. **Lesson 1: Introduction to R**)?

File: reference.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+`na.omit ( )` # Returns the object with incomplete cases removed
+
+`complete.cases ( )`# elements which are complete cases
+
+
+**Lesson 2**
+
+`download.file( ) ` #download the CSV file to your computer
+`read.csv( ) ` # load csv file into R memory 
+`head( ) `# check the top (the first 6 lines) of data .frame
```
Comment (fmichonneau) [inline reference.md]: check the top (the first 6 lines) of an object including data frames

File: reference.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+`%in% ` # to test if a value is found in a vector
+`is.na (  ) `# test if there are missing values
+`na.omit ( )` # Returns the object with incomplete cases removed
+
+`complete.cases ( )`# elements which are complete cases
+
+
+**Lesson 2**
+
+`download.file( ) ` #download the CSV file to your computer
```
Comment (fmichonneau) [inline reference.md]: -> download files from the internet to your computer

File: reference.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+`read.csv( ) ` # load csv file into R memory
```
Comment (fmichonneau) [inline reference.md]: csv -> CSV

File: reference.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+`rownames( ) ` # returns the row names
+`str( ) ` # check structure of the object and information about the class, length and content of each column
+`summary( ) `# summary statistics for each column
+
+`seq ( ) ` # generates a sequence of numbers
+
+
+**Lesson 4**
+
+`install.packages(“dplyr”) `
```
Comment (fmichonneau) [inline reference.md]: maybe just list `install.packages()` and give an explanation of what it does like the other entries of the glossary

File: reference.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+`library(“dplyr”) `
```
Comment (fmichonneau) [inline reference.md]: same here, just `library()` with an explanation of what it does.

File: reference.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+`seq ( ) ` # generates a sequence of numbers
+
+
+**Lesson 4**
+
+`install.packages(“dplyr”) `
+`library(“dplyr”) ` 
+`select ( ) `# select columns of a data frame,
+`filter ( ) `# allows you to select a subset of rows in a data frame 
+`%>% ` # pipes to select and filter at the same timeo
```
Comment (fmichonneau) [inline reference.md]: timeo -> time

File: reference.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+`%>% ` # pipes to select and filter at the same timeo 
+`mutate( ) ` #create new columns based on the values in existing columns
+`group_by( ) `# split the data into groups, apply some analysis to each group, and then combine the results.
+`summarize ( ) ` #collapses each group into a single-row summary of that group 
+`tally( )` # counts the total number of records for each category.
+`write.csv ( ) ` # save CSV file 
+
+**Lesson 5**
+
+`ggplot2(data=    , aes(x=  , y=))+geom_point( )+facet_wrap ( )+theme_bw( )+theme( ) `
```
Comment (fmichonneau) [inline reference.md]: could you please break down this call to explain what each function does?

File: reference.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+
+**Lesson 6**
+
+`dbConnect( ) `  # connect to the database
+
+`dbGetQuery (  ) ` # run the query and access data returned in a data.frame object.
```
Comment (fmichonneau) [inline reference.md]: data.frame -> data frame

File: reference.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+`read.csv( ) ` # load csv file into R memory 
+`head( ) `# check the top (the first 6 lines) of data .frame
+`factor( ) `# create factors
+`levels ( ) `# check levels of a factor
+`nlevels ( ) `# check number of levels of a factor
+`as.numeric(levels( )) ` # convert factors where the levels appear as numbers  to a numeric vector 
+
+**Lesson 3** 
+
+`data.frame ( )`# create data.frame
```
Comment (fmichonneau) [inline reference.md]: create a data.frame -> create a data frame

File: reference.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+`dim ( ) ` #check dimension of data frame
+`nrow ( ) ` # returns the number of rows
+`ncol ( ) `  # returns the number of  columns 
+`head ( ) ` # shows the first 6 rows
+`tail ( ) ` #shows the last 6 rows
+`names( ) ` #returns the column names (synonym of colnames() for data.frame objects)
```
Comment (fmichonneau) [inline reference.md]: data.frame -> data frame

**Comment by fmichonneau** [comment]
@aurielfournier do you have any other comments or suggestions for @bojiutk ?

**Comment by fmichonneau** [comment]
Hi @bojiutk,

I'm just checking in. Is there anything you'd need from us to be able to complete this?

Thanks!

**Comment by bojiutk** [comment]
if aurielfournier  doesn't have any other comments, I will do it soon. Just need a little more time. Too much to catch up on before holidays.

**Comment by fmichonneau** [comment]
Thanks! that sounds good

**Comment by bojiutk** [comment]
Hi @fmichonneau, I edited the file. Could you please check if the new reference.MD file is all right? I am not sure if I put it at the right place. Thanks!

**Comment by fmichonneau** [comment]
Hi @bojiutk, it doesn't look like the file with your latest edits has been committed to the repo yet.

**Comment by bojiutk** [comment]
Don't know how to deposit it to the right place in a right way. Here it is. Could you please move the reference.md file from this link to where it is supposed to be? 
https://github.com/bojiutk/R-ecology-lesson

I need to delete this forked repository after you move it. 

Thanks!

**Comment by fmichonneau** [comment]
Thanks, it's now included in the lesson materials.

**Pull request closed without merging**